File: WebDriverTests_Csharp/Autotest.Core/Autotests.Utilities/WebTestCore/SystemControls/TextInput.cs
using NUnit.Framework;
using OpenQA.Selenium;

namespace Autotests.Utilities.WebTestCore.SystemControls
{
    public class TextInput : HtmlControl
    {
        public TextInput(By locator, HtmlControl container = null)
            : base(locator, container)
        {
        }

        public virtual void SetValueAndWait(string value)
        {
            Clear();
            element.SendKeys(value + Keys.Tab);
        }

        public void SetValue(string value)
        {
            Clear();
            element.SendKeys(value);
        }

        public void Clear()
        {
            element.Clear();
        }

        public void Copy()
        {
            element.SendKeys(Keys.Control + "a" + "c" + Keys.Control + Keys.Tab);
        }

        public void Paste()
        {
            Clear();
            element.SendKeys(Keys.Control + "v" + Keys.Control + Keys.Tab);
        }

        public void AssertMaxLength(int maxLength)
        {
            var longText = new string('1', 500);
            SetValueAndWait(longText);
            string description =
                FormatWithLocator(string.Format("Ожидание максимально допустимой длины {0} в элементе", maxLength));
            Assert.AreEqual(maxLength, GetText().Length, description);
        }
    }
}

File: WebDriverTests_Java/WebDriverTests/src/test/java/core/systemControls/TextInput.java
package core.systemControls;

import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import ru.yandex.qatools.allure.annotations.Step;

public class TextInput extends HtmlControl
{

    public TextInput(By locator)
    {
        this(locator, null);
    }

    public TextInput(By locator, HtmlControl container)
    {
        super(locator, container);
    }

    @Step("Вводим зачение {0}")
    public void SetValueAndWait(String value)
    {
        Clear();
        element.sendKeys(value + Keys.TAB);
    }

    @Step("Вводим зачение {0}")
    public final void SetValue(String value)
    {
        Waiter.WaitExeption(() -> Clear());
        Waiter.WaitExeption(() -> element.sendKeys(value), "Не удалось ввести текст");
    }

    public final void Clear()
    {
        element.clear();
    }

    @Step("Copy зачение {0}")
    public final void Copy()
    {
        element.sendKeys(Keys.CONTROL + "a" + "c" + Keys.CONTROL + Keys.TAB);
    }

    @Step("Paste зачение {0}")
    public final void Paste()
    {
        Clear();
        element.sendKeys(Keys.CONTROL + "v" + Keys.CONTROL + Keys.TAB);
    }
}

